src/frontends/pytorch/src/op/scatter.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/scatter_elements_update.hpp"
+#include "openvino/op/slice.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+using namespace ov::op;
+
+OutputVector translate_scatter(const NodeContext& context) {
+    // Out-of-place schema
+    // aten::scatter.value(Tensor self, int dim, Tensor index, Scalar value) -> Tensor:
+    // aten::scatter.src(Tensor self, int dim, Tensor index, Tensor src) -> Tensor:
+    // aten::scatter.reduce(Tensor self, int dim, Tensor index, Tensor src, *, str reduce) -> Tensor:
+    // aten::scatter.value_reduce(Tensor self, int dim, Tensor index, Scalar value, *, str reduce) -> Tensor:
+
+    // Inplace schema
+    // aten::scatter_.value(Tensor(a!) self, int dim, Tensor index, Scalar value) -> Tensor(a!):
+    // aten::scatter_.src(Tensor(a!) self, int dim, Tensor index, Tensor src) -> Tensor(a!):
+    // aten::scatter_.reduce(Tensor(a!) self, int dim, Tensor index, Tensor src, *, str reduce) -> Tensor(a!):
+    // aten::scatter_.value_reduce(Tensor(a!) self, int dim, Tensor index, Scalar value, *, str reduce) -> Tensor(a!):
+    num_inputs_check(context, 4, 5);
+    auto input = context.get_input(0);
+    auto dim = context.get_input(1);
+    auto index = context.mark_node(std::make_shared<v0::Convert>(context.get_input(2), element::i32));
+    auto src = context.get_input(3);
+
+    auto reduction = v12::ScatterElementsUpdate::Reduction::NONE;
+    auto input_num = context.get_input_size();
+    if (input_num > 4 && !context.input_is_none(input_num - 1)) {
```

Comment:
```suggestion
    if (!context.input_is_none(input_num - 1)) {
```
Don't really need to check number of inputs that is verified inside `input_is_none`

Reply:
We need to check if schema contains reduce parameter, without this check `input_num > 4`, in cases without reduce param, last input would be src and it would fail due to trying to cast it to string.
As for setting reduce input index directly instead of relying on last element, it seem that index depends on input combination, couldn't figure out exact rules how it depends on it, last element was easiest simplest solution